### PR TITLE
Update geostyler dependencies

### PR DESCRIPTION
**Pull request by KaiVolland**
## Description

This updates the geostyler dependencies.

Due to some changes in the `geostyler-style` there are some adjustments needed. (seems like we should have released a major version in geostyler-style).
So it introduces a `FunctionUtil` (more or less a copy of the [OlStyleUtil](https://github.com/geostyler/geostyler-openlayers-parser/blob/master/src/Util/OlStyleUtil.ts)) and adds some throws in the `FilterUtil`.

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

### CODE REVIEW

File: src/Util/FilterUtil.ts [hunk 1]
```diff
@@ -342,6 +357,9 @@ class FilterUtil {
       newFilter = newFilter as CombinationFilter;
       newFilter.push(addedFilter);
     } else {
+      if (!Array.isArray(newFilter)) {
+        throw new Error(`Can not add filter to filter ${rootFilter}. Root filter has to be an array.`);
```
Comment (hblitza) [inline src/Util/FilterUtil.ts:361]:
```suggestion
        throw new Error(`Cannot add filter to filter ${rootFilter}. Root filter has to be an array.`);
```

File: src/Util/FilterUtil.ts [hunk 2]
```diff
@@ -372,6 +393,9 @@ class FilterUtil {
       case 'or':
         if (previousFilter && (previousFilter[0] === '&&' || previousFilter[0] === '||' )) {
           addedFilter = previousFilter;
+          if (!Array.isArray(addedFilter)) {
+            throw new Error('Can not change filter. Filter is not an array.');
```
Comment (hblitza) [inline src/Util/FilterUtil.ts:397]:
```suggestion
            throw new Error('Cannot change filter. Filter is not an array.');
```

**Comment by hblitza** [comment]
LGTM in general.

**Review by hblitza — APPROVED** [review]

File: src/Util/FunctionUtil.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+  T extends GeoStylerUnknownFunction ? any :
+  never;
+
+/**
+ * @class FunctionUtil
+ */
+class FunctionUtil {
+
+  public static evaluateFunction<T extends GeoStylerFunction>(func: T, feature?: Feature): ReturnType<T> {
+    if (func.name === 'property') {
```
Comment (ahennr) [inline src/Util/FunctionUtil.ts:56]: Is a null check for `func` needed?

File: src/Util/FunctionUtil.ts [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    }
+    const args = func.args.map(arg => {
+      if (isGeoStylerFunction(arg)) {
+        return FunctionUtil.evaluateFunction(arg, feature);
+      }
+      return arg;
+    });
+    switch (func.name) {
+      case 'abs':
+        return Math.abs(args[0] as number);
```
Comment (ahennr) [inline src/Util/FunctionUtil.ts:135]: What happens if no number is passed in the arguments? Should the args be checked here, e.g. using `_isFinite(…)`?

Comment (KaiVolland) [reply src/Util/FunctionUtil.ts:135]: This should not be a problem. This should just return `NaN` when you pass in an illegal `GeoStylerNumberFunction`. Classy shit in shit out. :smile:

**Pull request merged**